src/irr/asset/CGraphicsPipelineLoaderMTL.cpp

```diff
@@ -0,0 +1,1089 @@
+#include "irr/asset/CGraphicsPipelineLoaderMTL.h"
+
+namespace
+{
+    constexpr const char* VERT_SHADER_NO_UV_MAIN = 
+R"(
+void main()
+{
+    LocalPos = vPos;
+    gl_Position = irr_glsl_pseudoMul4x4with3x1(CamData.params.MVP, vPos);
+    ViewPos = irr_glsl_pseudoMul3x4with3x1(CamData.params.MV, vPos);
+    mat3 normalMat = irr_glsl_SBasicViewParameters_GetNormalMat(CamData.params);
+    Normal = normalMat*normalize(vNormal);
+}
+)";
+    constexpr const char* FRAG_SHADER_NO_UV_MAIN =
+R"(#define Ia 0.1
+void main()
+{
+    vec3 N = normalize(Normal);
+    vec3 L = normalize(-ViewPos);
+    vec3 R = -reflect(L,N);
+    float NdotL = max(dot(N,L), 0.0);
+    float VdotR = max(dot(L,R), 0.0);
+
+    vec3 color;
+    if (PC.params.extra&ILLUM_MODEL_MASK > 0)
+    {
+        color = PC.params.Ka*Ia + PC.params.Kd*NdotL;
```

Comment:
you need to build the `brdf_cos_eval` out of normalized BRDFs, so ones where `BRDF*cos(theta)` integrated over a hemisphere sums up to 1.0

Reply:
the `Kd` needs to be divided by PI and two functions need to be added to our engine-global BRDF headers
- `lambertian_cos_eval_rec_pi_factored_out`
- `lambertian_cos_eval`

```glsl
float lambertian_cos_eval_rec_pi_factored_out(....params)
{
   return params.NdotL;
}

float lambertian_cos_eval(....params)
{
   return lambertian_cos_eval_rec_pi_factored_out(params)*IRR_CORE_RECIPROCAL_PI;
}
```